esmvaltool/preprocessor/_mask.py

```diff
@@ -17,20 +19,81 @@
 logger = logging.getLogger(__name__)
 
 
-def mask_landocean(cube, land_mask=None, ocean_mask=None):
-    """Mask land or ocean"""
-    if land_mask or ocean_mask:
-        raise NotImplementedError
-
+def _get_fx_mask(fx_data, fx_option):
+    """Build a 50 percent land or ocean mask"""
+    inmask = np.zeros_like(fx_data, bool)
+    if fx_option == 'land':
+        # Mask land out
+        inmask[fx_data <= 50.] = False
+        inmask[fx_data > 50.] = True
+    elif fx_option == 'ocean':
+        # Mask ocean out
+        inmask[fx_data <= 50.] = True
+        inmask[fx_data > 50.] = False
+    return inmask
+
+
+def _apply_fx_mask(fx_mask, var_data):
+    """Apply the fx mask"""
+    var_mask = np.zeros(var_data.shape, bool)
+
+    # TIME-LAT-LON
```

Comment:
There is no need to treat all the number of dimensions separately, you can make use of [numpy broadcasting](https://docs.scipy.org/doc/numpy-1.13.0/user/basics.broadcasting.html).